cpp/include/raft/linalg/reduce_cols_by_key.cuh

```diff
@@ -52,6 +54,54 @@ void reduce_cols_by_key(const T* data,
 {
   detail::reduce_cols_by_key(data, keys, out, nrows, ncols, nkeys, stream);
 }
+
+/**
+ * @defgroup reduce_cols_by_key Reduce Across Columns by Key
+ * @{
+ */
+
+/**
+ * @brief Computes the sum-reduction of matrix columns for each given key
```

Comment:
I guess I haven't noticed how confusing this was before. If i was a user looking at a function called `reduce_cols_by_key`, I'd be expecting that I'd pass my own reduction function in. However, if I were invoking such a function without a reduction function (where another overload were provided that accepts a lambda), I'd expect it to perform a sum by default. It doesn't look like we provide such an option, though. What makes things even more confusing to me is that the `reduce_rows_by_key` function performs a weighted sum. 

I'm not saying we need to rename or remove these functions, but it would be useful if we had a github issue (and a todo here in this file that references it) to allow generic reduction functions. That would make this much more useful and also allow it to be extended for generalized row-norm computations).

Reply:
Still waiting on todo / github issue here